node-addon-api/Napi/napi.h

```diff
@@ -131,6 +131,27 @@ struct NapiValueRef : JsValueRef {
     virtual bool GetBoolean(bool defVal) const override { return isNull() ? defVal : m_napiVal.ToBoolean(); }
     virtual int32_t GetInt(int32_t defVal) const override { return isNumeric() ? m_napiVal.ToNumber() : defVal; }
     virtual uint32_t GetUInt(uint32_t defVal) const override { return isNumeric() ? m_napiVal.ToNumber() : defVal; }
+    virtual int64_t GetInt64(int64_t defVal) const override {
+        if (isNumeric()) {
+            double number = m_napiVal.ToNumber();
+            if (number >= std::numeric_limits<int64_t>::min() && number <= std::numeric_limits<int64_t>::max()) 

```

Comment:
Don't you want to test against [Number.MIN/MAX_SAFE_INTEGER](https://github.com/iTwin/imodel-native/blob/76e41a24831d794d0b382057909c210d5dfc3c94/iModelCore/libsrc/jsoncpp/include/json/value.h#L320)? Any value outside of that range cannot be trusted, because it was lossily converted to a double.

Reply:
Yep, made that change thanks.